app/components/polls/questions/answers_component.rb

```diff
@@ -0,0 +1,28 @@
+class Polls::Questions::AnswersComponent < ApplicationComponent
+  attr_reader :question
+  delegate :can?, :current_user, :user_signed_in?, to: :helpers
+
+  def initialize(question)
+    @question = question
+  end
+
+  def already_answered?(answer)
+    user_answers.find_by(answer: answer.title).present?
+  end
+
+  def voted_before_sign_in?
+    user_answers.any? do |vote|
+      vote.updated_at < current_user.current_sign_in_at
+    end
+  end
+
+  def question_answers
+    question.question_answers
+  end
+
+  private
+
+    def user_answers
+      @user_answers ||= question.answers.by_author(current_user.id)
```

Comment:
```suggestion
      @user_answers ||= question.answers.by_author(current_user)
```

No need to provide the ID and, even though right now we only call `user_answers` in methods that are only called if there's a `current_user`, this change will also make this method work with changes that make `current_user` `nil` at this point.

Reply:
Done. Thanks